### PR TITLE
fix: clarity version of nested requirements

### PR DESCRIPTION
**Pull request by hugocaillard**
### Description

Fix: #1079 

Follow up of https://github.com/hirosystems/clarinet/pull/997
With the previous fix, the order of the requirements would matter, now we get the right clarity version for each requirement

### Checklist

- [ ] Tests added in this PR (if applicable)
Note: I didn't add unit tests for project with nested dependencies. But I added the Alex project (which also revealed the bug prior to 997) to the release playbook.

I think this PR makes this part of code slightly simpler but I still think it can be revamped to be simpler and more robust

### CODE REVIEW

**Comment by hugocaillard** [comment]
@obycode Adding you as an (optional) reviewer since your the original author you might something we could miss

File: components/clarinet-deployments/src/lib.rs [hunk 1]
```diff
@@ -350,14 +345,10 @@ pub async fn generate_default_deployment(
                     ))
                 }
             };
-            // Download the code
-            let (_source, epoch, clarity_version, _contract_location) =
-                requirements::retrieve_contract(&contract_id, &cache_location, &file_accessor)
```
Comment (hugocaillard) [inline components/clarinet-deployments/src/lib.rs:355]: the purpose of this PR is to call `retrieve_contract` for each item in the queue and not just here

File: components/clarinet-deployments/src/lib.rs [hunk 2]
```diff
@@ -322,11 +321,7 @@ pub async fn generate_default_deployment(
                     ))
                 }
             };
-            queue.push_front((
-                contract_id,
-                StacksEpochId::Epoch21,
-                ClarityVersion::Clarity2,
-            ));
+            queue.push_front((contract_id, Some(DEFAULT_CLARITY_VERSION)));
```
Comment (obycode) [inline components/clarinet-deployments/src/lib.rs:324]: Do we not need the epoch anymore with this change?

Comment (hugocaillard) [reply components/clarinet-deployments/src/lib.rs:324]: @obycode It was already not need, it was only passed from items to items in the queue, but except for that, the value was never used

Comment (obycode) [reply components/clarinet-deployments/src/lib.rs:324]: Ah, I see. Thanks!

**Review by obycode — DISMISSED** [review]
LGTM!

**Review by lgalabru — APPROVED** [review]
thanks @hugocaillard! this part of the code is a bit sensitive and under documented.
what would be your take on adding a project in the `examples/`  folder, with some contracts outlining the scenario being fixed? The CI is running `clarinet check` on all this projects, and we would have an end to end test helping us with regressions.

**Comment by hugocaillard** [comment]
Done @lgalabru

**Review by lgalabru — DISMISSED** [review]
Thanks @hugocaillard 🙏

**Pull request merged**